benches/bench.rs

```diff
@@ -113,36 +131,75 @@ fn bench_get_many(c: &mut Criterion) {
     });
 }
 
-fn bench_set_large_with_string(c: &mut Criterion) {
+fn bench_add_with_large_string(c: &mut Criterion) {
     let rt = Runtime::new().unwrap();
 
-    c.bench_function("set_large_with_string", |b| {
+    c.bench_function("add_with_large_string", |b| {
         b.to_async(&rt).iter_custom(|iters| async move {
             let mut client = setup_client().await;
             let large_payload = "a".repeat(LARGE_PAYLOAD_SIZE);
             let start = std::time::Instant::now();
             for _ in 0..iters {
                 let _ = client
-                    .set("large_foo", large_payload.as_str(), None, None)
+                    .add("large_foo", large_payload.as_str(), None, None)
                     .await;
             }
             start.elapsed()
         });
     });
 }
 
-fn bench_add_large_with_string(c: &mut Criterion) {
+fn bench_set_multi_small_strings(c: &mut Criterion) {
     let rt = Runtime::new().unwrap();
 
-    c.bench_function("add_large_with_string", |b| {
+    c.bench_function("set_multi_small_strings", |b| {
         b.to_async(&rt).iter_custom(|iters| async move {
             let mut client = setup_client().await;
+            let kv = vec![("key1", "value1"), ("key2", "value2"), ("key3", "value3")];
+            let start = std::time::Instant::now();
+            for _ in 0..iters {
+                let _ = client.set_multi(&kv, None, None).await;
+            }
+            start.elapsed()
+        });
+    });
+}
+
+fn bench_set_multi_with_100_large_string_values(c: &mut Criterion) {
+    let rt = Runtime::new().unwrap();
+
+    c.bench_function("set_multi_with_100_large_string_values", |b| {
+        b.to_async(&rt).iter_custom(|iters| async move {
+            let mut client = setup_client().await;
+
             let large_payload = "a".repeat(LARGE_PAYLOAD_SIZE);
+
+            let mut keys = Vec::with_capacity(100);
+            let key_strings: Vec<String> = (0..100).map(|i| format!("key{}", i)).collect();
+            keys.extend(key_strings.iter().map(|s| s.as_str()));
```

Comment:
I reworked this a little bit but the `String` allocation via `format!()` seems to yield borrowing or lifetime issues in all cases if I don't collect the keys into a `Vec` first.  I'm fine with that in this case, since this is just the setup for a benchmark.